apps/rehash.c

```diff
@@ -220,6 +222,20 @@ static int handle_symlink(const char *filename, const char *fullpath)
     if (*endptr != '\0')
         return -1;
 
+    if (remove_links) {
+        if (verbose)
+            BIO_printf(bio_out, "unlink %s\n",
+                    fullpath);
+        if (unlink(fullpath) < 0 && errno != ENOENT) {
+            BIO_printf(bio_err,
+                    "%s: Can't unlink %s, %s\n",
+                    opt_getprog(), fullpath, strerror(errno));
+            return -1;
+        } else {
+            return 0;
+        }
+    }
```

Comment:
This looks virtually identical to similar code in two other places in `do_dir`. Perhaps pulling this out into a separate function so it can be used from all 3 places might be a good idea to reduce the copy&paste code

Reply:
+1 to that

I am also thinking whether this should not be done only for dangling symlinks. However I am not sure what happens with the current openssl (without this patch) if you change the contents of a certificate file to be a different certificate with different name hash.